### PR TITLE
docs: Add CONTRIBUTING.md and CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by nielsenko**
- Add standard _Contributor Covenant Code of Conduct_.
- Add a friendly CONTRIBUTING.md

## Summary by CodeRabbit

- **Documentation**
  - Added a Code of Conduct outlining community standards and enforcement procedures.
  - Added contribution guidelines detailing setup instructions, development workflow, code style, testing practices, and pull request process.

### CODE REVIEW

**Comment by nielsenko** [comment]
@coderabbitai review

**Comment by nielsenko** [comment]
@coderabbitai review

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+- **`type`**: `feat` (new feature), `fix` (bug fix), `docs` (documentation), `style` (formatting, linting), `refactor`, `test`, `chore` (build changes, etc.).
+- **Example**: `feat(router): add support for wildcard routes`
+
+### Testing Guidelines
+
+Comprehensive testing is crucial for maintaining the quality and stability of Relic. All new features and bug fixes should be accompanied by tests. The project primarily uses the `package:test/test.dart` framework. If using mocks, `package:mockito` is preferred. Follow its conventions for mock generation and usage.
+
+**1. Structure and Naming:**
+
+- **Grouping:** Tests should be organized into `group()` blocks. The description of the group should clearly state the component or functionality being tested (e.g., `'Normalization Logic'`, `'Request Parsing'`).
```
Comment (SandPod) [inline CONTRIBUTING.md:114]: Suggestion: Is this accurate? Do we follow this guideline here? I think the `file` is the grouping context in Serverpod and the test description merely follows the Given, When, Then guidelines.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+          final actualOutput = processMyFeature(input);
+
+          // Assert: Verify the outcome
+          expect(actualOutput, equals(expectedOutput));
+        });
+      });
+    }
+    ```
+
+- **File Naming:** Test files must end with `_test.dart` (e.g., `my_feature_test.dart`) and should generally mirror the name of the file they are testing, placed in a corresponding directory structure under `test/`.
```
Comment (SandPod) [inline CONTRIBUTING.md:147]: Suggestion: This might mirror the file but also mirror the scenario we are in. For example `multiple_routes_test.dart` in a folder named `router`?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+**4. General Testing Principles:**
+
+- **Clarity and Readability:** Tests are documentation. They should be easy to understand. The Given-When-Then structure is key.
+- **Isolation:** Each test should verify a single, specific aspect of behavior. Avoid testing multiple unrelated things in one test.
+- **Repeatability:** Tests must be deterministic and produce the same results every time they are run.
+- **Thoroughness:** Aim for good test coverage. Consider:
+  - Happy paths (valid inputs, expected behavior).
+  - Edge cases (empty inputs, nulls, boundaries, extremes).
+  - Error conditions (invalid inputs, exceptions).
+- **Maintainability:** Write tests that are robust to minor, unrelated code changes in the SUT (System Under Test).
```
Comment (SandPod) [inline CONTRIBUTING.md:170]: Suggestion: It might be good to mention that the tests should focus on the public interface so that the internal code could be completely rewritten without modifying the tests.

Avoid testing implementation details and focus on the interface guarantees.

**Review by SandPod — APPROVED** [review]
LGMT! 👍 

But would like to see the "group" suggestion modified so that we reflect this with different files instead if it can not be reflected in a `Given` statement.

File: CONTRIBUTING.md (hunk 1 above)
Comment (nielsenko) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md (hunk 2 above)
Comment (nielsenko) [reply CONTRIBUTING.md]: Updated
